doc/debug-system.md

```diff
@@ -0,0 +1,393 @@
+System Bus Access (SBA) (Section 3.10) | Supported with generic 32 or 64 bit wide interface.
+Authentication                         | Not supported at this time.
+Non-intrusive access to core registers | Not supported at this time. All registers must be accessed through the program buffer.
+
+# Theory of Operations
+
+The functionality of the debug system is implemented according to the RISC-V Debug Specification.
+Refer to this document for more information.
+
+## Block Diagram
+
+![Block diagram of the debug system](debugsys_schematic.svg)
+
+## Debug Module Registers
+
+An external debugger performs all interaction with the Debug Module through a register interface, accessed over a dedicated bus, the Debug Module Interface (DMI).
+The registers are called "Debug Module Registers" and defined in the RISC-V Debug Specification, Section 3.14.
+Our implementation only provides a single Debug Module on the DMI bus mapped to base address 0, hence all addresses below can be considered absolute.
+
+**Address** | **Name**                                     | **Implementation Notes**
+----------- | -------------------------------------------- | ------------------------
+0x04        | Abstract Data 0 (data0)
+0x0f        | Abstract Data 11 (data11)
+0x10        | Debug Module Control (dmcontrol)             | see table below
+0x11        | Debug Module Status (dmstatus)               | see table below
+0x12        | Hart Info (hartinfo)
+0x13        | Halt Summary 1 (haltsum1)
+0x14        | Hart Array Window Select (hawindowsel)
+0x15        | Hart Array Window (hawindow)
+0x16        | Abstract Control and Status (abstractcs)
+0x17        | Abstract Command (command)
+0x18        | Abstract Command Autoexec (abstractauto)
+0x19        | Configuration String Pointer 0 (confstrptr0)
+0x1a        | Configuration String Pointer 1 (confstrptr1)
+0x1b        | Configuration String Pointer 2 (confstrptr2)
+0x1c        | Configuration String Pointer 3 (confstrptr3)
+0x1d        | Next Debug Module (nextdm)
+0x1f        | Custom Features (custom)
+0x20        | Program Buffer 0 (progbuf0)
+0x2f        | Program Buffer 15 (progbuf15)
+0x30        | Authentication Data (authdata)
+0x32        | Debug Module Control and Status 2 (dmcs2)
+0x34        | Halt Summary 2 (haltsum2)
+0x35        | Halt Summary 3 (haltsum3)
+0x37        | System Bus Address 127:96 (sbaddress3)
+0x38        | System Bus Access Control and Status (sbcs)
+0x39        | System Bus Address 31:0 (sbaddress0)
+0x3a        | System Bus Address 63:32 (sbaddress1)
+0x3b        | System Bus Address 95:64 (sbaddress2)
+0x3c        | System Bus Data 31:0 (sbdata0)
+0x3d        | System Bus Data 63:32 (sbdata1)
+0x3e        | System Bus Data 95:64 (sbdata2)
+0x3f        | System Bus Data 127:96 (sbdata3)
+0x40        | Halt Summary 0 (haltsum0)
+0x70        | Custom Features 0 (custom0)
```

Comment:
I'm I out of the loop I don't see this is `v0.13.2` of the debug spec?

Reply:
Oh, that was added in 0.14. Removed it from our table.